src/Microsoft.Extensions.FileSystemGlobbing/Abstractions/InMemoryDirectoryInfo.cs

```diff
@@ -0,0 +1,134 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Microsoft.Extensions.FileSystemGlobbing.Abstractions
+{
+    public class InMemoryDirectoryInfo : DirectoryInfoBase
+    {
+        private static readonly char[] DirectorySeparators = new[] { Path.DirectorySeparatorChar, Path.AltDirectorySeparatorChar };
+        private readonly List<string> _files;
+
+        public InMemoryDirectoryInfo(string rootDir, List<string> files)
+        {
+            if (files == null)
+            {
+                _files = new List<string>();
+            }
+            else
+            {
+                _files = files;
+
+                // normalize
+                for (int i = 0; i < _files.Count; ++i)
+                {
+                    _files[i] = Path.GetFullPath(_files[i].Replace(Path.AltDirectorySeparatorChar, Path.DirectorySeparatorChar));
+                }
+            }
+
+            FullName = Path.GetFullPath(rootDir.Replace(Path.AltDirectorySeparatorChar, Path.DirectorySeparatorChar));
+            Name = Path.GetFileName(rootDir);
+        }
+
+        // This is to avoid the overhead of Path.GetFullPath from the public constructor
+        private InMemoryDirectoryInfo(string rootDir, List<string> files, bool normalized)
```

Comment:
Check normalized argument and do normalization logic in this constructor, call passing false from default one.